src/components/contact-govuk-one-login/contact-govuk-one-login-controller.ts

```diff
@@ -1,7 +1,29 @@
 import { Request, Response } from "express";
+import { isValidUrl } from "./../../utils/strings";
+import pino from "pino";
 
-export function contactGet(req: Request, res: Response): void {
-  const data = {};
+const CONTACT_ONE_LOGIN_TEMPLATE = "contact-govuk-one-login/index.njk";
+const logger = pino();
 
-  res.render("contact-govuk-one-login/index.njk", data);
+export function contactGet(req: Request, res: Response): void {
+  const fromUrl = getFromUrlAndSaveIt(req);
+  if (!fromUrl) {
+    logger.error(
+      "Request to contact-govuk-one-login page did not contain a valid fromUrl in the request or session"
+    );
+  }
+  const data = {
+    fromUrl: fromUrl,
+  };
+  res.render(CONTACT_ONE_LOGIN_TEMPLATE, data);
 }
+
+const getFromUrlAndSaveIt = (request: Request): string => {
+  const fromUrlFromRequest = request.query.fromUrl as string;
```

Comment:
```suggestion
  const fromUrlFromRequest = request.query.fromURL as string;
```
We've asked teams to call the parameter `fromURL` (note the capitals). I don't think we need to bother changing the name elsewhere in our code